tests/conftest.py

```diff
@@ -10,122 +11,235 @@
+                    assert len(g.members) > 0
+                    c.test_history.append(self.make_message(
+                        g, c, g.members[mi % len(g.members)],
+                        f'Mock message {mi} in #mock-channel-{ci}'))
+
+    def queue_event(self, event_name, *args, **kwargs):
+        """Queues an event to be sent to registered listeners."""
+
+        if not event_name.startswith('on_'):
+            raise ValueError(f"event '{event_name}' doesn't start with 'on_'")
+        self.event_queue.append((event_name, args, kwargs))
+
+    async def async_dispatch_events(self):
+        """Sends all queued events to registered handlers."""
+
+        while self.event_queue:
+            batch, self.event_queue = self.event_queue, []
+            for event_name, args, kwargs in batch:
+                handler = getattr(self.context.client, event_name, None)
+                if handler is not None:
+                    await handler(*args, **kwargs)
+
+    def sim_reaction(self, message, unicode, user, delta):
+        """Simulates an emoji reaction changes and queues notification events.
+
+        Args:
+            message - the message object to modify
+            unicode - unicode of emoji to add/remove
+            user - the user adding/removing the emoji
+            delta - +1 to add, -1 to remove
+        """
+
+        if delta not in (-1, +1):
+            raise ValueError('reaction {unicode} delta {delta} not -1 or +1')
+
+        message_reaction = next(
+            (r for m in message.reactions if str(r.emoji) == unicode), None)
+        if message_reaction is None:
+            message_reaction = self.pytest_mocker.Mock(
+                spec=discord.Reaction, name='reaction')
+            message_reaction.emoji = self.pytest_mocker.MagicMock(
+                spec=discord.PartialEmoji, name='reaction.emoji')
+            message_reaction.emoji.name = unicode
+            message_reaction.emoji.__str__.return_value = unicode
+            message_reaction.count = 0
+            message_reaction.me = False
+            message_reaction.message = message
+            message.reactions.append(message_reaction)
+        if message_reaction.count + delta < 0:
+            raise ValueError('reaction {unicode} count dropped below 0')
+        if user.id == self.context.client.user.id:
+            message_reaction.me = (delta > 0)
+        message_reaction.count += delta
+
+        event = self.pytest_mocker.Mock(
+            spec=discord.RawReactionActionEvent, name='raw_reaction_event')
+        event.message_id = message.id
+        event.user_id = user.id
+        event.channel_id = message.channel.id
+        event.guild_id = message.guild.id
+        event.emoji = message_reaction.emoji
+        event.member = user
+        event.event_type = f'REACTION_{"ADD" if delta > 0 else "REMOVE"}'
+        self.queue_event(
+            f'on_raw_reaction_{"add" if delta > 0 else "remove"}', event)
```

Comment:
I find these interpolated if-elses with double-quoted strings kinda hard to read, and also a little redundant. Possible alternative:
```python
action = "ADD" if delta > 0 else "REMOVE"
event.event_type = f'REACTION_{action}'
self.queue_event(f'on_raw_reaction_{action.lower()}', event)
```
("action" might not be the best name for this, but you get the gist)

Reply:
Simplified!